File: app/src/main/java/msk/pobazar/wcquiz/presenter/MenuPresenter.java
package msk.pobazar.wcquiz.presenter;

import com.arellomobile.mvp.InjectViewState;
import com.arellomobile.mvp.MvpPresenter;
import com.arellomobile.mvp.presenter.ProvidePresenter;

import msk.pobazar.wcquiz.view.MenuView;


@InjectViewState
public class MenuPresenter extends MvpPresenter<MenuView> {
    private MenuView view;

    @ProvidePresenter
    MenuPresenter newInstance() {
        return new MenuPresenter();
    }

    @Override
    protected void onFirstViewAttach() {
        super.onFirstViewAttach();
    }

    @Override
    public void attachView(MenuView view) {
        super.attachView(view);
        this.view = view;
    }
}

File: feature_menu/src/main/java/msk/pobazar/wcquiz/feature_menu/presenter/MenuPresenter.kt
package msk.pobazar.wcquiz.feature_menu.presenter

import moxy.InjectViewState
import msk.pobazar.wcquiz.core.base.BasePresenter
import msk.pobazar.wcquiz.core.navigation.Router
import msk.pobazar.wcquiz.core.navigation.screens.NavigationScreen
import msk.pobazar.wcquiz.core.navigation.transitionsParams.GameParams
import javax.inject.Inject

@InjectViewState
class MenuPresenter @Inject constructor(
    private val router: Router
) : BasePresenter<MenuView>() {
    fun onQuickGameClick() {
        router.show(NavigationScreen.Game(GameParams()))
    }

    fun onModsClick() {
        router.show(NavigationScreen.Mods)
    }

    fun onRatingClick() {
        router.show(NavigationScreen.Rating)
    }
}